File: lab_8_SQL_FULL_ACCESS/mcp-server/server-stdio.js
/**
 * PostgreSQL MCP server over stdio (for local / subprocess clients).
 * Run: node server-stdio.js — communicates via stdin/stdout.
 */
import "dotenv/config";
// Use console.error so logs go to stderr and appear in parent (stdout is used for MCP protocol)
import { StdioServerTransport } from "@modelcontextprotocol/sdk/server/stdio.js";
import { server } from "./mcp-server.js";
const transport = new StdioServerTransport(process.stdin, process.stdout);

server.onerror = (err) => {
  console.error("[postgres-mcp stdio error]", err);
};

await server.connect(transport)
// Process stays alive; stdin is read for JSON-RPC messages.

File: lab_8_SQL_FULL_ACCESS_PY/mcp_server/server_stdio.py
"""
PostgreSQL MCP server over stdio (for local / subprocess clients).
Run: python -m mcp_server.server_stdio — communicates via stdin/stdout.
"""
if __name__ == "__main__":
    from dotenv import load_dotenv
    load_dotenv()

    from mcp_server.server import mcp
    mcp.run(transport="stdio")
